tests/Utils/SchemaExtenderTest.php

```diff
@@ -1962,4 +1963,48 @@ public function testOriginalResolversArePreserved()
         $result = GraphQL::executeQuery($extendedSchema, $query);
         self::assertSame(['data' => ['hello' => 'Hello World!']], $result->toArray());
     }
+
+
+    /**
+     * @see https://github.com/webonyx/graphql-php/issues/180
+     */
+    public function testShouldBeAbleToIntroduceNewTypesThroughExtension()
+    {
+        $sdl = '
+          type Query {
+            defaultValue: String
+          }
+          type Foo {
+            value: Int
+          }
+        ';
+
+        $documentNode = Parser::parse($sdl);
+        $schema       = BuildSchema::build($documentNode);
+
+        $extensionSdl = '
+          type Bar {
+            foo: Foo
+          }
+        ';
+
+        $extendedDocumentNode = Parser::parse($extensionSdl);
+        $extendedSchema       = SchemaExtender::extend($schema, $extendedDocumentNode);
+
+        $expected = '
+            type Bar {
+              foo: Foo
+            }
+            
+            type Foo {
+              value: Int
+            }
+            
+            type Query {
+              defaultValue: String
+            }
+        ';
+
+        static::assertEquals($this->dedent($expected), SchemaPrinter::doPrint($extendedSchema));
```

Comment:
maybe simply `self::` would do? I guess `static::` works too but `self` is more common and I don't see a reason for `static`